### DISCUSSION TITLE
why not all metrics are collected ArgoCD?

### DISCUSSION THREAD

**Discussion by Iceforestik** [argoproj/argo-cd#10911] [Q&A]
I set the nodePort value for argocd-metrics and argocd-server-metrics services, but for some reason the metrics specified in the documentation are not given? for example, the argocd_app_info metric or argocd_app_labels, there is only argocd_redis_request_duration_bucket etc are given.

**Comment by jgwest** [comment]
They should all be available: you may want to verify that your k8s network configuration is correctly serving each of the metrics endpoints of each of the pods. You may want to use an `Ingress` for this, for example, rather than a `NodePort`. For general info about exposing Argo CD endpoints is [here](https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/).

**Comment by Iceforestik** [comment]
@jgwest hi, tell me, please, I made a forward port and the metrics were outstanding. for what reason can not be given by node port? do you have any ideas how to debug?
I can confirm that everything worked through ingress, but this is not really an option for me. Why nodeport doesn't start.
